packages/react/src/components/Table/TableToolbar/TableToolbarAdvancedFilterFlyout.jsx

```diff
@@ -309,7 +311,39 @@ const TableToolbarAdvancedFilterFlyout = ({
                       return options;
                     };
                     const memoizeColumnOptions = memoize(filterColumnOptions); // TODO: this memoize isn't really working, should refactor to a higher column level
-
+                    if (column.isDate) {
+                      return (
+                        <FormItem
+                          key={`simple-filter-item-${rowIndex}-${columnIndex}`}
+                          className={`${iotPrefix}--filter-flyout__simple-field`}
+                        >
+                          <DatePicker
+                            key={`${columnIndex}-${columnFilterValue}`}
+                            datePickerType="single"
+                            dateFormat="Y-m-d"
```

Comment:
We should allow for a way to customize the `dateFormat` and add a `locale`. Perhaps we can add to the `options` prop type to include an object for settings.

Reply:
As the `options` prop is specific for the "select" fields, I created a dateOptions object to customize the `dateFormat` and `locale`.

Reply:
As the options prop is specific for the "select" fields, I created a dateOptions object to customize the dateFormat and locale.